### PR TITLE
[#49] Expose proper error message for the login API

### LINKED ISSUE #49: Expose proper error message for the login API
## Why

In the survey login api, when the email/password are wrong, It is returning  errors response like this 
Screen Shot 2020-12-21 at 11 42 05 AM

it is returning same errors messages when the client id/secret is wrong, also it returns 400  status code,  it should be 401 

In Braive project we return like this
Screen Shot 2020-12-18 at 5 23 16 PM, 
We can follow Braive errors message pattern here.

### PR DESCRIPTION
**Pull request by hoangmirs**
Resolve #49

## What happened

- Customize password grant flow to return the correct error message.
- Create the custom classes and re-register the grant flow.
- Fix tests

 
## Insight

N/A
 

## Proof Of Work

Postman

### CODE REVIEW

File: spec/controllers/api/v1/tokens_controller_spec.rb [hunk 1]
```diff
@@ -38,39 +38,52 @@
...
           post :create, params: params
 
-          response_body = JSON.parse(response.body)
+          expected_response = {
+            errors: [
+              {
+                code: 'invalid_email_or_password',
+                detail: 'Your email or password is incorrect. Please try again.',
+                source: {
+                  parameter: 'Doorkeeper::OAuth::Error'
```
Comment (olivierobert) [inline spec/controllers/api/v1/tokens_controller_spec.rb:73]: This reveals a bit too much of the internals of the application so it is not great in terms of security 🙈

**Review by olivierobert — COMMENTED** [review]
What is the response status in case of error? Is it 401 or 200?

File: spec/controllers/api/v1/tokens_controller_spec.rb (hunk 1 above)
Comment (hoangmirs) [reply spec/controllers/api/v1/tokens_controller_spec.rb]: Yes, and I think the current `source` parameter is not correct, so I removed them in https://github.com/nimblehq/nimble-survey-web/pull/68/commits/aa70ecc36cabbeefdf714e391fdf90546010a622

**Comment by hoangmirs** [comment]
> What is the response status in case of error? Is it 401 or 200?

Currently, it returns `400 - Bad Request`. That was the default from Doorkeeper, I didn't change it

File: spec/controllers/api/v1/tokens_controller_spec.rb (hunk 1 above)
Comment (hoangmirs) [reply spec/controllers/api/v1/tokens_controller_spec.rb]: @olivierobert Removed them in https://github.com/nimblehq/nimble-survey-web/pull/68/commits/aa70ecc36cabbeefdf714e391fdf90546010a622

**Review by olivierobert — APPROVED** [review]

**Review by suho — APPROVED** [review]
LGTM

**Pull request merged**